src/vs/workbench/services/textfile/electron-browser/textFileService.ts

```diff
@@ -55,20 +56,73 @@ export class TextFileService extends AbstractTextFileService {
 	}
 
 	public resolveTextContent(resource: URI, options?: IResolveContentOptions): TPromise<IRawTextContent> {
-		return this.fileService.resolveStreamContent(resource, options).then(streamContent => {
-			return ModelBuilder.fromStringStream(streamContent.value).then(res => {
-				const r: IRawTextContent = {
-					resource: streamContent.resource,
-					name: streamContent.name,
-					mtime: streamContent.mtime,
-					etag: streamContent.etag,
-					encoding: streamContent.encoding,
-					value: res.value,
-					valueLogicalHash: res.hash
-				};
-				return r;
+
+		const result: IRawTextContent = {
+			resource: undefined,
+			name: undefined,
+			mtime: undefined,
+			etag: undefined,
+			encoding: undefined,
+			value: undefined,
+			valueLogicalHash: undefined,
+		};
+
+		let contentToken = new CancellationTokenSource();
+		let completePromise: Thenable<any>;
+
+		let statsPromise = this.fileService.resolveFile(resource).then(stat => {
```

Comment:
I think this has to move into the `resolveStringStream` method because otherwise the `FILE_NOT_MODIFIED_SINCE` error will never be thrown for anyone that is using this method outside of the `textFileService`. If no `etag` is provided, the `stat` can run in parallel to the `fs.open` in `resolveStringStream` and otherwise we have to run it first.

The stat could then also be used within `resolveStringStream` to cancel reading the file in chunks if the file size is detected to be too large.